pkg/client/shub/util.go

```diff
@@ -54,19 +54,16 @@ func shubParseReference(src string) (uri ShubURI, err error) {
 	} else if l < 2 {
 		return ShubURI{}, errors.New("Not a valid Shub reference")
 	}
-
-	// look for an @ and split if it exists
-	if strings.Contains(src, `@`) {
-		refParts = strings.Split(src, `@`)
-		uri.digest = `@` + refParts[1]
+	// look for a : and split if it exists
+	if strings.Contains(src, ":") {
```

Comment:
this is doing the work twice: once to look for ":" and a second time to look for ":" again and split the string.

It's equivalent to something like this:

```
if parts := strings.SplitN(src, ":", 2); len(parts) > 1 {
    src = parts[0]
    url.tag = parts[1]
}
```